### DISCUSSION TITLE
Design an NFT minting contract

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#89] [Questions] [answered]
Asked by `kjeih` in discord
>Hey community.
I am wondering about nft collection deployment.
Actually, I deployed a nft minting smart contract.
I attached a feature for minting several nft collections on this contract.
But someone said me it is not user-friendly.
He said, users are checking nft minting transaction from one smart contract.
So, he said nft minting transaction should be created by using unique collection address as target address.
This means, I should create nft smart contract for every collection. and should mint nfts on this unique smart contract.
Is that right?

**Comment by 0x-j** [comment, marked as answer]
I think you are trying to build a launchpad contract that serves as a registry for all your collections and an entry point for user to mint? If so I think your current design is the standard design. 

But if you are trying to customize the behavior of each collection, like adding additional metadata to NFT, then you need to deploy contract per collection. [This contract](https://github.com/aptos-labs/aptogotchi/blob/main/move/sources/aptogotchi.move) is an example of we customize the collection by adding additional metadata, so every Aptogotchi collection is deployed with a new contract.

**Reply by MoonShiesty** [reply]
> then you need to deploy contract per collection

i think a better design would be to have the launchpad/collection creator as a single contract, so there is a single registry of all collections, and then each project launching on the launchpad can implement their own contract/metadata which extends the collection to customize.

ie the launchpad contract creates the collection, passes ownership to the caller, and returns an extend_ref to the caller so they can customize the NFT collection with their own metadata

**Reply by 0x-j** [reply]
Yeah that'd be even better, a launchpad supports custom collections that still act as an entry point for all mints.
